File: Count_Asterisks.java
class Solution {
    public int countAsterisks(String s) {
        int count = 0;
        int helper = 0;
        for(int i = 0;i<s.length();i++){
            if(s.charAt(i)== '|'){
                helper++;
            }
          if(s.charAt(i)== '*'&&helper%2==0){
              count++;
          }
        }
   return count; }
}

File: Count_Asterisks.cpp
class Solution {
public:
    int countAsterisks(string s) {
        int ans=0;
        int p=0;
        for(char c:s){
            if(c=='|')p++;
            ans+=(p%2==0 && c=='*');
        }
        return ans;
    }
};
